substratevm/src/com.oracle.objectfile/src/com/oracle/objectfile/debugentry/ClassEntry.java

```diff
@@ -99,7 +100,8 @@ public ClassEntry(String className, FileEntry fileEntry, int size) {
         this.fileEntry = fileEntry;
         // methods is a sorted list and we want to be able to add more elements to it while keeping
         // it sorted,
-        // so a LinkedList seems more appropriate than an ArrayList. (see getMethodEntry)
+        // so a LinkedList seems more appropriate than an ArrayList.
+        // (see ensureMethodEntryForDebugRangeInfo)
```

Comment:
How is using a `LinkedList` more appropriate than an `ArrayList`?

It's the opposite. While `ArrayList` can use an optimized sort (`java.util.ArrayList#sort` performing an `Arrays.sort` directly on the underlying `elementData` array, `LinkedList` has to use the fallback implementation that first turns the elements into a temporary array that then gets sorted and then gets written back into the `LinkedList` via ListIterator.

Reply:
Please grep through the whole debuginfo sourcebase and replace all occurrences of LinkedList with ArrayList. If you find a place where you really have to use LinkedList for a good reason please let me know.

Reply:
I fount `LinkedList` more appropriate because despite the less efficient sort, it allows us to perform a sorted insertion in https://github.com/oracle/graal/pull/3419/files/a0cf3be29aaa5b635aa2ace659c39f976f4f56e0#diff-b7692d10d2dc550e4bbf155d2f7f142dbdaca4aca41c2677d41b193149c168a4R366 without having to shift elements.
Now, given that adding more methods to the `ClassEntry` during code processing and thus needing sorted insertions appears to be rare (empirically), the cost of shifting the elements might indeed be not significant, so I think we could go with `ArrayList` as well.

Reply:
Resolved in https://github.com/oracle/graal/pull/3419/commits/6bd40df1bf617ed2532fd9136d8daf67448f4bac